ui/src/hooks/useMerkleTreeWorker.ts

```diff
@@ -11,18 +11,10 @@ export const useMerkleTreeWasmWorker = () => {
   ): Promise<Uint8Array> => {
     process.env.NODE_ENV === 'development' && console.time('==>merkle')
 
-    try {
-      const proof = await MerkleTreeWorker.generateMerkleProof(
-        leaves,
-        leaf,
-        depth,
-      )
+    const proof = await MerkleTreeWorker.generateMerkleProof(leaves, leaf, depth)
 
-      process.env.NODE_ENV === 'development' && console.timeEnd('==>merkle')
-      return proof
-    } catch (error) {
-      throw error
-    }
+    process.env.NODE_ENV === 'development' && console.timeEnd('==>merkle')
+    return proof
```

Comment:
did you test whether this works?
I remember telling you myself this is a useless try /catch (just rethrowing in the catch block is useless), but when I removed it the UI wasn't rendering the error correctly. Especially the error boundary component/page (see [error.tsx](https://github.com/anonklub/anonklub/tree/main/ui/src/app/prove/submit) would not catch it and present a default screen instead).

Can you check if we still have the correct behavior? That is: an error/throw in
`await MerkleTreeWorker.generateMerkleProof(
        leaves,
        leaf,
        depth,
      )`
      
 should trigger the rendering of the [error component](https://github.com/anonklub/anonklub/blob/main/ui/src/app/prove/submit/error.tsx)